### PR TITLE
Fix #537: Add routine to produce JWT by signing claims with private key

### LINKED ISSUE #537: Add routine to produce JWT by signing claims with private key
We already sign any data, so why not claims in order to directly produce JWT.

### PR DESCRIPTION
**Pull request by petrdvorak**

### CODE REVIEW

File: proj-xcode/PowerAuth2/PowerAuthSDK.m [hunk 1]
```diff
@@ -1474,6 +1474,40 @@ - (NSData*) generateInvalidBiometricKey
...
+                                         data:claimsData
+                                     callback:^(NSData * _Nullable signature, NSError * _Nullable error) {
+        // Handle error
+        if (error) {
+            callback(nil, error);
+            return;
+        }
+        
+        // Prepare JWT Header
+        NSDictionary *header = [NSDictionary dictionaryWithObjects:@[@"ES256", @"JWT"] forKeys:@[@"alg", @"typ"]];
```
Comment (hvge) [inline proj-xcode/PowerAuth2/PowerAuthSDK.m:1493]: I would prefer a more readable form here:
```
@{ @"alg": @"ES256", @"typ": @"JWT" }
```

**Review by hvge — CHANGES_REQUESTED** [review]
Looks OK, I would just suggest you to use more modern ObjC constructs while creating constant dictionaries. 

What I miss is Android implementation.

File: proj-xcode/PowerAuth2/PowerAuthSDK.m (hunk 1 above)
Comment (petrdvorak) [reply proj-xcode/PowerAuth2/PowerAuthSDK.m]: I am an $#&*$, sure this looks way better...

**Comment by hvge** [comment]
I have added Android implementation and integration tests.

**Review by hvge — APPROVED** [review]

**Pull request merged**